### PR TITLE
new WORKING machine - Namennayo (Japan)

### PR DESCRIPTION
**Pull request by DavidHaywood**
new WORKING machine
------
Namennayo (Japan) [ShouTime, David Haywood]

### CODE REVIEW

File: src/mame/drivers/galaxian.cpp [hunk 1]
```diff
@@ -5309,6 +5327,56 @@ static INPUT_PORTS_START( scramble )
 	PORT_BIT( 0xff, 0x00, IPT_UNUSED )
 INPUT_PORTS_END
 
+// unknown button config, button 1 moves up the screen, with up being jump and down being drop item
+// using standard mapping, but maybe that is correct?
```
Comment (angelosa) [inline src/mame/drivers/galaxian.cpp:5331]: Galaxian HW has interchangeable input pinouts for Up and Fire cfr. [here](https://www.mikesarcade.com/cgi-bin/spies.cgi?action=url&type=pinout&page=Pac.html)
If I've understood what the game is correctly it makes more sense the other way around?

**Review by angelosa — CHANGES_REQUESTED** [review]

File: src/mame/drivers/galaxian.cpp (hunk 1 above)
Comment (DavidHaywood) [reply src/mame/drivers/galaxian.cpp:5331]: I don't know, it seems easier to play with up as jump and down as drop bomb, moving up the screen is more like accelerate, connected to a button, I *think* it's correct as is.

Comment (angelosa) [reply src/mame/drivers/galaxian.cpp:5331]: I totally don't expect jump to be up on a racing game. 
And to make a comparison: Bump N Jump doesn't map jump to up.

Comment (DavidHaywood) [reply src/mame/drivers/galaxian.cpp:5331]: Bump N Jump is a different style of game.

Comment (angelosa) [reply src/mame/drivers/galaxian.cpp:5331]: Both are overhead racing games of the early 80s, and BnJ maps up as acceleration.

Comment (DavidHaywood) [reply src/mame/drivers/galaxian.cpp:5331]: ShouTime had the instruction / wiring sheet anyway, Up is jump, Down is release Item, A is accelerate http://mamedev.emulab.it/haze/reference/namenayo_instructions.jpg so MAME is correct

Comment (angelosa) [reply src/mame/drivers/galaxian.cpp]: I don't see Up clearly written there.

Comment (DavidHaywood) [reply src/mame/drivers/galaxian.cpp]: Then you can't read.

Comment (angelosa) [reply src/mame/drivers/galaxian.cpp]: If it's confirmed then update the PORT_NAME labels reflecting it (it still says `// move up screen` which is confusing as hell).

**Comment by angelosa** [comment]
I locked conversation cause author keeps resolving what I'm saying without doing his homework first.

I request that the Up and Down directions have a "Pn Up / Jump" / "Pn Down / Bomb" **PORT_NAME** attached to them as also stated in my last comment. I'm gonna merge when that happens.

**Comment by cuavas** [comment]
@DavidHaywood is correct, we’re discouraging these kinds of descriptions in the `PORT_NAME` besides cases where they’re absolutely essential.  That’s what the port name plugin is for.

**Comment by angelosa** [comment]
This is literally the first time I hear that `PORT_NAME` is deprecated de-facto. Please explain.

**Comment by DavidHaywood** [comment]
PORT_NAME is primarily for 'non-obvious' eg. control pads with all sorts of labelled buttons where a list of unnamed buttons would be difficult to map without names.

For cases like this PORT_NAME is harmful as it obfuscates what the actual controls were.

**Comment by DavidHaywood** [comment]
Personally I think we should start maintaining files for said plugin, and including them as part of the distribution, and possibly even start move dipswitch naming external too.

**Comment by angelosa** [comment]
Is anything even committed data-wise on upstream?

**Comment by angelosa** [comment]
@DavidHaywood leftmost column has wrong priority over player sprite (submerged).
0000x
0001x
Pretty much game breaking imho, do you have a quick fix?

**Comment by DavidHaywood** [comment]
hmm, I don't think I ever went all the way to the left in the whole time I tested / played it.

the fix is that 
	clip.min_x = std::max(clip.min_x, (!m_flipscreen_x) * (16 + hoffset) * m_x_scale);
	clip.max_x = std::min(clip.max_x, (256 - m_flipscreen_x * (16 + hoffset)) * m_x_scale - 1);

in the sprite draw doesn't apply here, although will need to be made custom somehow.

let me see what the cleanest way to go about that is.

**Comment by DavidHaywood** [comment]
https://github.com/mamedev/mame/pull/7688

**Pull request merged**